lib/github/ldap.rb

```diff
@@ -180,10 +183,10 @@ def search(options, &block)
       instrument "search.github_ldap", options.dup do |payload|
         result =
           if options[:base]
-            @connection.search(options, &block)
+            forest_search(options, &block)
           else
             search_domains.each_with_object([]) do |base, result|
-              rs = @connection.search(options.merge(:base => base), &block)
+              rs = forest_search(options.merge(:base => base), &block)
               result.concat Array(rs) unless rs == false
             end
```

Comment:
Never seen the tree (```options[:base] == empty```) to be called. The loop logic to iterate over ```search_domains``` seems to be always implemented outside the library within GHE